Assets/MixedRealityToolkit.Extensions/SceneTransitionService/Scripts/CameraFaderQuad.cs

```diff
@@ -0,0 +1,204 @@
+                    Debug.LogException(e);
+                    return;
+                }
+            }
+
+            quadMaterial.SetColor(QuadMaterialColorName, currentColor);
+
+            // Create our quads
+            foreach (Camera camera in targets)
+            {
+                // Can't target the same camera twice
+                if (quads.ContainsKey(camera))
+                    continue;
+
+                Quad quad = new Quad();
+                quad.PropertyBlock = new MaterialPropertyBlock();
+                quad.Renderer = GameObject.CreatePrimitive(PrimitiveType.Quad).GetComponent<Renderer>();
+                quad.Renderer.sharedMaterial = quadMaterial;
+                // Parent the quad under the camera
+                quad.Renderer.transform.parent = camera.transform;
+                quad.Renderer.transform.localScale = camera.orthographic ? Vector3.one * camera.orthographicSize : Vector3.one * camera.fieldOfView;
+                quad.Renderer.transform.localPosition = Vector3.forward * camera.nearClipPlane * 1.01f;
+                quad.Renderer.transform.localRotation = Quaternion.identity;
+
+                // Set the quad's layer to something the camera sees
+                for (int layer = 0; layer < 32; layer++)
+                {
+                    if (camera.cullingMask == (camera.cullingMask | (1 << layer)))
+                    {
+                        quad.Renderer.gameObject.layer = layer;
+                        break;
+                    }
+                }
+
+                quads.Add(camera, quad);
+            }
+
+            // Perform our fade
+            float fadeAmount = 0;
+            while (fadeAmount < 1)
+            {
+                fadeAmount += Time.unscaledDeltaTime;
+                currentColor = Color.Lerp(fadeInColor, fadeOutColor, fadeAmount);
+
+                foreach (Quad quad in quads.Values)
+                {
+                    // Must have been destroyed - just continue
+                    if (quad.Renderer == null)
+                        continue;
+
+                    quad.PropertyBlock.SetColor(QuadMaterialColorName, currentColor);
+                    quad.Renderer.SetPropertyBlock(quad.PropertyBlock);
+                }
+
+                await Task.Yield();
+            }
+
+            await Task.Yield();
+
+            State = CameraFaderState.Opaque;
+        }
+
+        public async Task FadeInAsync(float fadeInTime)
+        {
+            if (quads.Count == 0)
+            {
+                Debug.LogError("No camera targets found - are you trying to fade in before you've faded out?");
+                return;
+            }
+
+            switch (State)
+            {
+                case CameraFaderState.Opaque:
+                    break;
+
+                default:
+                    Debug.LogWarning("Can't fade in in state " + State + " - not proceeding.");
+                    break;
+            }
```

Comment:
This is a nit, but I've seen it a couple of times and was hoping to adopt a simpler thing - if you only have a single state you're checking for (i.e. ! = Opaque) just use an if statement, it's simpler in that case. Switches are handy if you're trying to show how logic branches off when multiple options come into play.